lib/ansible/modules/cloud/digital_ocean/do_floating_ip.py

```diff
@@ -0,0 +1,329 @@
+        status_code = response.status_code
+        json = response.json
+        if status_code == 204:
+            module.exit_json(changed=True)
+        elif status_code == 404:
+            module.exit_json(changed=False)
+        else:
+            module.exit_json(changed=False, data=json)
+
+
+def get_floating_ip_details(module, rest):
+    ip = module.params['ip']
+
+    response = rest.get("floating_ips/{}".format(ip))
+    status_code = response.status_code
+    json = response.json
+    if status_code == 200:
+        return response.json['floating_ip']
+    else:
+        module.fail_json(msg="Error assigning floating ip [{}: {}]".format(
+            status_code, response.json["message"]), region=module.params['region'])
+
+
+def assign_floating_id_to_droplet(module, rest):
+    ip = module.params['ip']
+
+    payload = {
+        "type": "assign",
+        "droplet_id": module.params['droplet_id'],
+    }
+
+    response = rest.post("floating_ips/{}/actions".format(ip), data=payload)
+    status_code = response.status_code
+    json = response.json
+    if status_code == 201:
+        wait_action(module, rest, ip, json['action']['id'])
+
+        module.exit_json(changed=True, data=response.json)
+    else:
+        module.fail_json(msg="Error creating floating ip [{}: {}]".format(
+            status_code, response.json["message"]), region=module.params['region'])
+
+
+def associate_floating_ips(module, rest):
+    floating_ip = get_floating_ip_details(module, rest)
+    droplet = floating_ip['droplet']
+
+    # TODO: If already assigned to a droplet verify if is one of the specified as valid
+    if droplet is not None and str(droplet['id']) in [module.params['droplet_id']]:
+        module.exit_json(changed=False)
+    else:
+        assign_floating_id_to_droplet(module, rest)
+
+
+def create_floating_ips(module, rest):
+    payload = {
+    }
+
+    if module.params['region'] is not None:
+        payload["region"] = module.params['region']
+    if module.params['droplet_id'] is not None:
+        payload["droplet_id"] = module.params['droplet_id']
+
+    response = rest.post("floating_ips", data=payload)
+    status_code = response.status_code
+    json = response.json
+    if status_code == 202:
+        module.exit_json(changed=True, data=response.json)
+    else:
+        module.fail_json(msg="Error creating floating ip [{}: {}]".format(
+            status_code, response.json["message"]), region=module.params['region'])
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            ip=dict(aliases=['id'], required=False),
+            region=dict(required=False),
+            droplet_id=dict(required=False),
+            oauth_token=dict(
+                no_log=True,
+                # Support environment variable for DigitalOcean OAuth Token
+                fallback=(env_fallback, ['DO_OAUTH_TOKEN']),
```

Comment:
Other DO modules are already looking at "DO_API_TOKEN", "DO_API_KEY". Should we keep this standard as we move forward so users don't have to setup more than one environment variable?

Please see above comments about the use of oauth_token.